src/lib/mina/account-update.ts

```diff
@@ -1056,6 +1071,22 @@ class AccountUpdate implements Types.AccountUpdate {
     }
     return accountUpdate;
   }
+
+  /**
+   * Create an account update that is added to the transaction only if a condition is met.
+   *
+   * See {@link AccountUpdate.create} for more information. In this method, you can pass in
+   * a condition that determines whether the account update should be added to the transaction.
+   */
+  static createIf(condition: Bool, publicKey: PublicKey, tokenId?: Field) {
+    return AccountUpdate.create(
+      // if the condition is false, we use an empty public key, which causes the account update to be ignored
```

Comment:
not the protocol, just o1js. it will skip "dummy" account updates both when hashing the public input:
https://github.com/o1-labs/o1js/blob/6c52fea324cbe37607c88d03908157149165dac8/src/lib/mina/account-update.ts#L1540-L1542

and when forming the transaction:
https://github.com/o1-labs/o1js/blob/6c52fea324cbe37607c88d03908157149165dac8/src/lib/mina/account-update.ts#L1597-L1598

it's a bespoke system of enabling optional account updates that I added at some point because it was needed for something